### ISSUE TITLE
Version 1.0 milestones

### ISSUE THREAD

**Issue by jjhelmus** [jjhelmus/pyfive#1]
This issue will be used to track work towards the `1.0` milestone where pyfive can read all HDF5 files which are supported by hdf5 using a similar API.  This issue will be updates as new issues are opened.

HDF5 features not supported by pyfive

* compression
* shuffling
* complex attributes
* various datatypes
* dimension scale
* unlimited dimensions
* multiple unlimited dimensions
* version 1.10 files

h5py API features not implemented in pyfive

* `visit` methods.
* `visititems` methods.
* `read_direct` Dataset method.
* `astype` Dataset method.
* **chunks** Dataset attribute.
* **compression** Dataset attribute.
* **compression_opts** Dataset attribute.
* **scaleoffset** Dataset attribute.
* **shuffle** Dataset attribute.
* **fletcher32** Dataset attribute.
* **fillvalue** Dataset attribute.

**Comment by bnlawrence** [comment]
Given that the discussion on #89 suggests that it will be quite a while before we implement that, and given that with the eventual completion of #114 close, my thought is that we are ready for V1.0 now, and that the issues around support for scale-offset (and anything else) should be pushed into "post V1.0".  I think the most active folks with an opinion on this would be @bmaranville, @kmuehlbauer, @davidhassell  and @valeriupredoi. What say you all? Remove scaleoffset from here, and aim for a V1.0 with #95 ?

**Comment by valeriupredoi** [comment]
one thing I reckon we can still work on before v1.0 is test coverage; @kmuehlbauer would you reckon pre-commit/ruff support would be something we want before v1.0 also? 🍻

**Comment by kmuehlbauer** [comment]
@valeriupredoi Yes, if we transform the codebase using the latest listing and formatting, this would indeed be wanted for a version 1.

I found another glitch with REFERENCE type over at h5netcdf, which would be great to sort out before V1.0. I'll open an issue in the next day describing the problem. Maybe there is a quick solution to this.

**Comment by valeriupredoi** [comment]
> [@valeriupredoi](https://github.com/valeriupredoi) Yes, if we transform the codebase using the latest linting and formatting, this would indeed be wanted for a version 1.
> 
> I found another glitch with REFERENCE type over at h5netcdf, which would be great to sort out before V1.0. I'll open an issue in the next day describing the problem. Maybe there is a quick solution to this.

eagle eye 🦅 
will start the motions for modern linting with pre-commit+ruff then 🍺
